src/Foundation/NSOperatingSystemVersion.cs

```diff
@@ -36,20 +36,102 @@ namespace Foundation  {
 	[SupportedOSPlatform ("macos")]
 	[SupportedOSPlatform ("tvos")]
 #endif
-	// NSProcessInfo.h
-	public struct NSOperatingSystemVersion {
+
+	public struct NSOperatingSystemVersion: IEquatable<NSOperatingSystemVersion>, IComparable{
 		public nint Major, Minor, PatchVersion;
 		
-		public NSOperatingSystemVersion (nint major, nint minor, nint patchVersion)
+        public NSOperatingSystemVersion (nint majorVersion, nint minorVersion, nint patchVersionArg)
+		{
+			Major = majorVersion;
+			Minor = minorVersion;
+			PatchVersion = patchVersionArg;
+		}
+
+        public NSOperatingSystemVersion (nint majorVersion, nint minorVersion)
```

Comment:
This can be done by chaining to the more full constructor:
```
public NSOperatingSystemVersion (nint majorVersion, nint minorVersion)
    : this (majorVersion, minorVersion, 0);
{
}
```